### PR TITLE
Add openPMD-api

### PR DESCRIPTION
**Pull request by ax3l**
Add a recipe for openPMD-api.

Repo: https://github.com/openPMD/openPMD-api
Homepage: https://www.openPMD.org
License: LGPL 3.0+

### CODE REVIEW

**Comment by ax3l** [comment]
@C0nsultant I am adding you as a co-maintainer for this conda-forge package. Is this ok with you?

**Comment by C0nsultant** [comment]
Certainly. You probably have more expertise with the binding, I can provide more background if something behin the curtain goes wrong.

File: recipes/openpmd-api/build.sh [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+cmake \
+    -DCMAKE_BUILD_TYPE=Release  \
+    -DopenPMD_USE_MPI=OFF       \
+    -DopenPMD_USE_HDF5=ON       \
+    -DopenPMD_USE_ADIOS1=ON     \
+    -DopenPMD_USE_ADIOS2=OFF    \
+    -DopenPMD_USE_PYTHON=ON     \
+    -DPYTHON_EXECUTABLE:FILEPATH=$(which $PYTHON)      \
+    -DBUILD_TESTING=OFF                                \
```
Comment (ax3l) [inline recipes/openpmd-api/build.sh]: can enable later on in the repo as soon as https://github.com/openPMD/openPMD-api/pull/227 is fixed and released

File: recipes/openpmd-api/build.sh [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+cmake \
+    -DCMAKE_BUILD_TYPE=Release  \
+    -DopenPMD_USE_MPI=OFF       \
+    -DopenPMD_USE_HDF5=ON       \
+    -DopenPMD_USE_ADIOS1=ON     \
+    -DopenPMD_USE_ADIOS2=OFF    \
+    -DopenPMD_USE_PYTHON=ON     \
+    -DPYTHON_EXECUTABLE:FILEPATH=$(which $PYTHON)  \
+    -DBUILD_TESTING=OFF             \
```
Comment (ax3l) [inline recipes/openpmd-api/build.sh]: can enable later on in the repo as soon as https://github.com/openPMD/openPMD-api/pull/227 is fixed and released

**Comment by ax3l** [comment]
@conda-forge/staged-recipes ready for review :-)

File: recipes/openpmd-api/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+  host:
+    - python
+    - pybind11 >=2.2.1
+    - cmake >=3.10
+    - adios >=1.13.1  # [unix]
+    - hdf5  >=1.8.13
+  run:
+    - python
+    - adios >=1.13.1  # [unix]
+    - hdf5  >=1.8.13
```
Comment (scopatz) [inline recipes/openpmd-api/meta.yaml]: Does this not need numpy as a run req.?

File: recipes/openpmd-api/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+    This conda-forge package only provides serial I/O, please build from source
+    for MPI parallel I/O on HPC systems. Window support is limited to HDF5 and
+    Python 3.
+  doc_url: https://openpmd-api.readthedocs.io
+  dev_url: https://github.com/openPMD/openPMD-api
+
+extra:
+  recipe-maintainers:
+    - ax3l
+    - C0nsultant
```
Comment (scopatz) [inline recipes/openpmd-api/meta.yaml]: @C0nsultant - are you OK with being a maintainer?

**Comment by scopatz** [comment]
Thanks for putting this in @ax3l!

File: recipes/openpmd-api/meta.yaml (hunk 4 above)
Comment (ax3l) [reply recipes/openpmd-api/meta.yaml]: yes he confirmed above :) https://github.com/conda-forge/staged-recipes/pull/6032#issuecomment-395050766

File: recipes/openpmd-api/meta.yaml (hunk 3 above)
Comment (ax3l) [reply recipes/openpmd-api/meta.yaml]: not yet, in later releases it will!

**Pull request merged**